src/utils/filterResults.ts

```diff
@@ -37,13 +38,37 @@ function getOwnerData (property: UnprocessedResultsFromCRM) {
     return ownerData
 }
 
-export default function filterResults (unsortedPropertyResults: UnprocessedResultsFromCRM[], searchParameters: SearchParametersType[]): { matchedProperties: UnprocessedResultsFromCRM[], uniqueSearchRecords: string[] } {
+export default function filterResults (unsortedPropertyResults: UnprocessedResultsFromCRM[], searchParameters: IntersectedSearchAndFilterParams[], filterInUse: string): { matchedProperties: UnprocessedResultsFromCRM[], uniqueSearchRecords: string[] } {
+    let desiredPropertyTypes = searchParameters[0].propertyTypes
+    let desiredPropertyGroups = searchParameters[0].propertyGroups
+    let isPropertyTypeFilterInUse = desiredPropertyTypes.length !== 0
```

Comment:
why is this needed? 

`isUnderPropertyTypeLimit && matchForPropertyTypes(property, desiredPropertyTypes)`

would always be false if desiredPropertyTypes is []

Reply:
According to Chris's spec he wants:
1) Default: nothing entered into anywhere >> return all records
2) If either property type or property group is used >> disable the one not used and return properties that have the value of the type or group entered. They always use OR logic even if used together.
3) If property managed is used with any combination of type or group >> return the combination and if the property is managed. The managed is uses AND logic with group or type.

I've made the default state for group and type to an empty array so I know they don't want to use that filter and it disables it by making max results zero. But if none of them are used them then I'm assuming they want to return all properties.

Reply:
I don't think you need a variable for isPropertyTypeFilterInUse or any of the extra if statements that were added.

I think they already worked independently the way we had it before. 

For example, if the property type filter is disabled, 

```isUnderPropertyTypeLimit && matchForPropertyTypes(property, desiredPropertyTypes)``` will always return false because `desiredPropertyTypes` would be `[]` and `matchForPropertyTypes` would also return false because 

![image](https://user-images.githubusercontent.com/1034890/88264528-3b38c080-cd0f-11ea-9685-8e03eb74b4a3.png)


RE default state: shouldn't the default state be All for all filters?